src/node_sqlite.cc

```diff
@@ -825,13 +825,21 @@ void DatabaseSync::ApplyChangeset(const FunctionCallbackInfo<Value>& args) {
       Local<Function> filterFunc = filterValue.As<Function>();
 
       filterCallback = [env, filterFunc](std::string item) -> bool {
+        TryCatch try_catch(env->isolate());
         Local<Value> argv[] = {String::NewFromUtf8(env->isolate(),
-                                                   item.c_str(),
-                                                   NewStringType::kNormal)
-                                   .ToLocalChecked()};
-        Local<Value> result =
-            filterFunc->Call(env->context(), Null(env->isolate()), 1, argv)
-                .ToLocalChecked();
+                                                  item.c_str(),
+                                                  NewStringType::kNormal)
+                                  .ToLocalChecked()};
+        MaybeLocal<Value> maybe_result =
```

Comment:
I don't think you need the `TryCatch` here. The problem is the use of `ToLocalChecked()`. Take a look at [this code](https://github.com/nodejs/node/blob/69502d8835bd02cd7f52a981f5674535e63caf72/src/node_sqlite.cc#L202-L207). You can tell if V8 has an exception pending if the `ToLocal()` call does not succeed.